apps/src/applab/firebaseLogging.js

```diff
@@ -0,0 +1,45 @@
+import logToCloud from '../logToCloud';
+
+const LOG_DATA_TRANSFER_INTERVAL_MS = 1000;
+
+/**
+ * The timeout ID for the next flush of data transfer logs, if one is pending.
+ * @type {number}
+ */
+let flushDataTransferTimeout = null;
+
+/**
+ * The number of bytes transferred since the last flush.
+ * @type {number}
+ */
+let firebaseByteCount = 0;
+
+/**
+ * The number of write operations since the last flush.
+ * @type {number}
+ */
+let firebaseWriteCount = 0;
+
+/**
+ * Records that the specified number of bytes have been transferred to or from Firebase.
+ * This data is logged to NewRelic via addPageAction. A maximum of 20 page actions can
+ * be generated every 10 seconds, therefore we use batching and send the total number
+ * of bytes transferred at most once per second.
+ * @param {number} byteCount
```

Comment:
Sending data once per second seems like a lot just for logging purposes. Doesn't this mean we're going to be making a bunch of largely unnecessary network requests? I wonder if we can get away with doing this a lot less frequently.

Does firebase give us no way to get similar metrics itself? Seems a little silly to have the client doing a bunch of writes to a second service to track how many writes we're doing to the first service.